components/org.wso2.carbon.identity.entitlement.proxy/src/main/org/wso2/carbon/identity/entitlement/proxy/PEPProxyFactory.java

```diff
@@ -79,7 +79,12 @@ private PEPProxyFactory(){
                 if (!serverUrl.endsWith("/")) {
                     serverUrl += "/";
                 }
-
+                if (appConfig.get(USER_NAME) == null || appConfig.get(USER_NAME).length() == 0) {
```

Comment:
possible NPE in appConfig. If appId is not in appToPDPClientConfigMap, appConfig can be null